src/Orleans.Runtime/GrainDirectory/GrainDirectoryHandoffManager.cs

```diff
@@ -196,6 +186,12 @@ internal async Task ProcessSiloStoppingEvent(DirectoryMembershipSnapshot members
             Dictionary<GrainId, IGrainInfo> batchUpdate = localDirectory.DirectoryPartition.GetItems();
 
             await HandoffMyPartitionUponStop(batchUpdate, silosHoldingMyPartitionCopy, true);
+
+            // This will cause remote write requests to be forwarded to the silo that will become the new owner.
+            // Requests might bounce back and forth for a while as membership stabilizes, but they will either be served by the
+            // new owner of the grain, or will wind up failing. In either case, we avoid requests succeeding at this silo after we've
+            // completed handoff, which could cause them to not get handed off to the new owner.
+            this.HasPerformedHandoff = true;
```

Comment:
Should this comment be added to https://github.com/dotnet/orleans/pull/5737/files#diff-eaa3e506dd98e5155c01b308f0dd4962R33 too?